doc/api/url.md

```diff
@@ -129,6 +129,13 @@ return `true`.
 
 #### `new URL(input[, base])`
 
+<!--
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/47339
+    description: ICU requirement is removed.
```

Comment:
Aren't virtually all official / semi-official builds using full ICU, or at least small ICU? This seems like a technical detail that most users won't care about, especially since it's about _removing_ a requirement, so any existing setup will keep working.

Reply:
Yes, all existing setups will keep working, but what do you think is the correct way to communicate that our non-icu builds will have proper hostname parsing?

Reply:
I think @tniessen point is that none of our builds are non-icu. This only affects people who compile Node.js themselves.

Reply:
I see. If you think it's not worth the `notable-change` label, feel free to remove it.

Reply:
Sure, we can treat this as a notable change and add YAML changes blocks to all relevant APIs, and I won't object to that. It just feels insignificant enough that most people won't care -- the vast majority of users use official builds that always support ICU, and those who don't still build with ICU by default.